### ISSUE TITLE
get demo download link

### ISSUE THREAD

**Issue by LiDesheng2333** [akiver/csgo-sharecode#76]
How to use this to get demo download link 。

**Comment by akiver** [comment]
Demo links are located inside the last entry of the `roundstatsall` field of the protobuf message [CDataGCCStrike15_v2_MatchInfo](https://github.com/SteamDatabase/GameTracking-CS2/blob/master/Protobufs/cstrike15_gcmessages.proto#L921) (it's the field `map`).

The protobuf message is stored inside a `.info` file next to the demo when you download it from the game.
A share code doesn't contain this information, but it allows you to retrieve the `CDataGCCStrike15_v2_MatchInfo` message from the Steam Game Coordinator.
You can use [boiler-writter](https://github.com/akiver/boiler-writter) to retrieve it and then parse the protobuf message to get the demo link.
You could take a look at how it's done in [CS Demo Manager](https://github.com/akiver/CSGO-Demos-Manager/blob/main/Services/Concrete/DemosService.cs#L425).

To sum up:
1. Decode the share code to get the values of `matchId`, `outcomeId` and `tokenId`
2. Spawn boiler-writter to write the proto message in a file: `./boiler-writter data.info matchId outcomeId tokenId`
3. Parse the message inside `data.info`. The language is up to you, if it's JS you could use [csgo-protobuf](https://github.com/akiver/csgo-protobuf). There is an example in the readme.
4. The demo link will be in the `map` field of the last entry of `roundstatsall`
